src/ucp/rma/flush.c

```diff
@@ -32,22 +32,35 @@ static int ucp_ep_flush_is_completed(ucp_request_t *req)
 
 static void ucp_ep_flush_progress(ucp_request_t *req)
 {
-    ucp_ep_h ep = req->send.ep;
+    ucp_ep_h       ep        = req->send.ep;
+    ucp_lane_map_t all_lanes = UCS_MASK(ucp_ep_num_lanes(ep));
     ucp_ep_flush_state_t *flush_state;
     ucp_lane_index_t lane;
     ucs_status_t status;
     uct_ep_h uct_ep;
+    int diff;
+
+    /* If the number of lanes changed since flush operation was submitted, adjust
+     * the number of expected completions */
+    if (req->send.flush.num_lanes != ucp_ep_num_lanes(ep)) {
+        diff                            = ucp_ep_num_lanes(ep) -
+                                          req->send.flush.num_lanes;
+        req->send.state.uct_comp.count += diff;
+        req->send.flush.num_lanes       = ucp_ep_num_lanes(ep);
+        ucs_trace_req("flush req %p: adjusting expected completion count by %d",
+                      req, diff);
+    }
 
-    ucs_trace("ep %p: progress flush req %p, lanes 0x%x count %d", ep, req,
-              req->send.flush.lanes, req->send.state.uct_comp.count);
+    ucs_trace("ep %p: progress flush req %p, started_lanes 0x%x count %d", ep,
+              req, req->send.flush.started_lanes, req->send.state.uct_comp.count);
 
-    while (req->send.flush.lanes) {
+    while (req->send.flush.started_lanes < all_lanes) {
```

Comment:
I don't think that it's correct to compare masks by `<`. 
Is it possible if number of lanes stays the same but some bit "migrates" to other position after EP reconfiguration?
I guess there are possible more exotic side effects.

If we assume that lanes can't be disabled after reconfiguration and the mask only grows after reconfiguration, pls add corresponding assert.

Reply:
the lanes are always contiguous (no "holes" in the array). I'll add assert that mask always grows.